### PR TITLE
[adding_type_declarations]: Adding type declarations to ux capture core lib

### PR DESCRIPTION
**Pull request by bstolpe**
This PR adds type declarations to UX capture core library by creating UXCaptureTypes file which stores all our types that can be imported.

Questions: 

- Is completeMark param on Line 73 & 84 in Zone being used or should it be removed?

### CODE REVIEW

**Comment by sergeychernyshev** [comment]
Hmmm. I feel we should do it differently - essentially, we need types to be consumable by external application that uses TypeScript rather than the types in the code of UX Capture which does not use TypeScript itself.

Here's an example of the types we created for React implementation at some point:
https://github.com/ux-capture/ux-capture/blob/main/packages/react-ux-capture/types/index.d.ts

So we need something similar for the core library.

File: packages/ux-capture/src/UXCapture.js [hunk 1]
```diff
@@ -106,7 +106,7 @@ const UXCapture = {
 	 *
 	 * @returns {object}|null
 	 */
-	getViewConfig: () => {
+	getViewConfig: () => {
```
Comment (TerryOShea) [inline packages/ux-capture/src/UXCapture.js:109]: did you mean to add a space here?

File: packages/ux-capture/src/View.js [hunk 2]
```diff
@@ -1,52 +1,53 @@
...
  * A `View` is a collection of zones representing a single
  * page view or interactive view.
  *
  * Refer to ux-capture README glossary for view definitions.
  * @see: https://github.com/ux-capture/ux-capture#glossary
  */
-function View(props) {
-	this.props = props;
-	this.expectedZones = this.setZones(this.props.zoneConfigs);
+class View {
```
Comment (TerryOShea) [inline packages/ux-capture/src/View.js:9]: Is refactoring this file necessary for adding types?

File: packages/ux-capture/types/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+export declare interface View extends ViewProps {
+    update: UpdateFunction;
+    setZones: SetZonesFunction;
+    destroy: DestroyFunction;
+    createZone: createZoneFunction;
+    getZoneConfigs: getZoneConfigsFunction;
+}
+
+export declare interface ExpectedMark extends ExpectedMarkProps {
+    _mark: MarkFunction;
```
Comment (TerryOShea) [inline packages/ux-capture/types/index.d.ts:49]: does the _ prefix mean that people shouldn't be accessing this property?

File: packages/ux-capture/types/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+export declare interface View extends ViewProps {
+    update: UpdateFunction;
+    setZones: SetZonesFunction;
+    destroy: DestroyFunction;
+    createZone: createZoneFunction;
+    getZoneConfigs: getZoneConfigsFunction;
+}
+
+export declare interface ExpectedMark extends ExpectedMarkProps {
```
Comment (TerryOShea) [inline packages/ux-capture/types/index.d.ts:48]: why is there both an ExpectedMark and an ExpectedMarkProps? what's the difference?

**Review by TerryOShea — COMMENTED** [review]
I would try to avoid doing too many unrelated refactors--they distract from the main purpose of this PR.

Will we also have to bump the version number in package.json?

I notice that the existing react types are written as types but these are interfaces--what's the difference there?

Also, there are some failing automated checks below that you'll need to look into before pushing:
Screen Shot 2022-03-28 at 1 22 33 PM

File: packages/ux-capture/src/View.js (hunk 2 above)
Comment (bstolpe) [reply packages/ux-capture/src/View.js:9]: No my typescript plugin seems to be applying auto suggestions without prompting me first

File: packages/ux-capture/types/index.d.ts (hunk 4 above)
Comment (bstolpe) [reply packages/ux-capture/types/index.d.ts:48]: I was assuming these object methods needed type support but now realizing that's not the case; will remove them and condense this down.

File: packages/ux-capture/src/View.js [hunk 5]
```diff
@@ -11,42 +10,41 @@ function View(props) {
 	this.props = props;
 	this.expectedZones = this.setZones(this.props.zoneConfigs);
 }
-
```
Comment (sergeychernyshev) [inline packages/ux-capture/src/View.js:14]: I don't think ypu meant to change this file at all, right?

File: packages/ux-capture/src/Zone.js [hunk 6]
```diff
@@ -1,5 +1,4 @@
 import ExpectedMark from './ExpectedMark';
-
```
Comment (sergeychernyshev) [inline packages/ux-capture/src/Zone.js:2]: Same goes for this one, right?

**Review by sergeychernyshev — CHANGES_REQUESTED** [review]
You can just exclude the changes for files that you only re-formatted.

File: packages/ux-capture/src/ExpectedMark.js [hunk 7]
```diff
@@ -84,7 +84,7 @@ ExpectedMark.prototype._mark = function () {
 	this.marked = true;
 
 	// call all registered zone callbacks
-	this.onMarkListeners.forEach(listener => listener(this));
+	this.onMarkListeners.forEach((listener) => listener(this));
```
[the remark this answers is not in the document]
Comment (bstolpe) [reply packages/ux-capture/src/ExpectedMark.js]: Most of these changes were from the typescript plugin auto formatting file so no these aren't necessary and have been reverted

**Review by sergeychernyshev — APPROVED** [review]

**Pull request merged**